src/DGtal/shapes/MeshVoxelizer.h

```diff
@@ -0,0 +1,226 @@
+  /**
+   Description of template class 'MeshVoxelizer'
+
+   \brief Aim: A class for computing the digitization of a triangle or
+   a Mesh.
+
+   This class is parametrized by a Separation strategy (either 6 or
+   26). This value corresponds to the separability of a digitized
+   continuous plane. For instance, if the value is set to 6, then
+   the digitization of an Euclidean plane is such that no 6-path
+   exists to go from one side to the other of the Euclidean plane (see
+   @cite KletteRosenfeld_book).
+   
+   From the separability parameter, a template is constructed. A voxel
+   belongs to the digitization if and only if the template centered at
+   a given voxel intersects the triangle.
+
+   This approach is a CPU straightforward implementation of
+   @cite Laine13.
+
+   6 and 26 templates are the following ones:
+
+   @image html 6-sep.png "Template for 6-separating digitization"
+   @image html 26-sep.png "Template for 26-separating digitization"
+
+
+   @tparam TDigitalSet a DigitalSet (model of concepts::CDigitalSet)
+   @tparam Separation strategy of the voxelization (6 or 26)
+   */
+  template <typename TDigitalSet, int Separation = 6>
+  class MeshVoxelizer
+  {
+
+  public:
+
+    ///Concept Checking
+    BOOST_CONCEPT_ASSERT(( concepts::CDigitalSet<TDigitalSet> ));
+
+    ///Digital Set Type
+    typedef TDigitalSet DigitalSet;
+    BOOST_STATIC_ASSERT_MSG( DigitalSet::Space::dimension == 3, "DigitalSet dimension must be 3");
+
+    ///Separation Checking
+    BOOST_STATIC_ASSERT_MSG( Separation == 6 || Separation == 26, "Separation must be 6 or 26");
+
+    /****** Associated types *********************/
+    using Space = typename DigitalSet::Space;
+    using Space2D = SpaceND<2, typename Space::Integer>;
+    using MeshFace = std::vector< unsigned int >;
+    using Domain   = typename DigitalSet::Domain;
+    using PointR3  = typename Space::RealPoint;
+    using VectorR3 = typename Space::RealPoint;
+    using PointR2  = typename Space2D::RealPoint;
+    using PointZ3  = typename Space::Point;
+    using OrientationFunctor = InHalfPlaneBySimple3x3Matrix<PointR2, double>;
+    using IntersectionTarget = typename IntersectionTargetTrait<Space, Separation, 1>::Type;
+    /*********************************************/
+
+  public:
+
+    /**
+     * Constructor of the voxelizer
+     */
+    MeshVoxelizer() = default;
+
+    // ----------------------- Standard services ------------------------------
+    /**
+     * Voxelize the mesh into the digital set.
+     * @warning if the mesh has non-trianuglar faces, we naively triangulate
+     * using a triangle fan at zero. If the face is not convex, the output
+     * will not be correct.
+     *
+     * @param [out] outputSet the set that collects the voxels.
+     * @param [in] aMesh the mesh to voxelize (vertex coordinates will
+     * be casted to @e PointR3 points.
+     * @param [in] scaleFactor the scale factor to apply to the mesh
```

Comment:
for this parameter perhaps you have to alter to be careful to not use a value compatible with the domain of the digital set, else we have an error... (or/and detect it explicitly later ?)

Reply:
what do you mean ?
That the scale may induce voxels outside the set domain ?

Reply:
sorry something was auto completed with my editor, or I was very tired ;)
I was thinking that you could add a  user warning  in the doc to avoid this potential error. But with the last change, we can simply mention that outside domain voxel are ignored ?